.github/workflows/tests.yaml

```diff
@@ -0,0 +1,36 @@
+# From https://docs.github.com/en/actions/automating-builds-and-tests/building-and-testing-nodejs#using-the-nodejs-starter-workflow
+name: Node.js and Solana CI
+
+on:
+  push:
+    branches: [main]
+  pull_request:
+    branches: [main]
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    strategy:
+      matrix:
+        node-version: [20.x]
+
+    steps:
+      - uses: actions/checkout@v3
+      # Install Solana
+      # From https://docs.solana.com/cli/install-solana-cli-tools
+      - run: sh -c "$(curl -sSfL https://release.solana.com/stable/install)"
```

Comment:
Check out web3.js for a strategy to cache the whole entire CLI install between runs.

Reply:
https://github.com/solana-labs/solana-web3.js/blob/94b4d494590b2ffbf60751bb74ed267b7228f012/.github/workflows/cicd.yml#L35-L50

Reply:
Legend! Have filed an issue, will add the caching next week.